src/middlewared/middlewared/plugins/system.py

```diff
@@ -524,27 +524,39 @@ async def dmidecode_info(self):
             self.DMIDECODE_CACHE['ecc-memory'] = ecc
 
         if self.DMIDECODE_CACHE['baseboard-manufacturer'] is None:
-            bm = ((await run(["dmidecode", "-s", "baseboard-manufacturer"], check=False)).stdout.decode(errors='ignore')).strip()
+            bm = (
+                (await run(["dmidecode", "-s", "baseboard-manufacturer"], check=False)).stdout.decode(errors='ignore')
+            ).strip()
             self.DMIDECODE_CACHE['baseboard-manufacturer'] = bm
 
         if self.DMIDECODE_CACHE['baseboard-product-name'] is None:
-            bpn = ((await run(["dmidecode", "-s", "baseboard-product-name"], check=False)).stdout.decode(errors='ignore')).strip()
+            bpn = (
```

Comment:
Is it useful to set all of these variables separately and have a separate `if` condition for each variable ? In other words, won't this only be true once ?

Reply:
Correct, this should only be set once for each item. This is static data that never changes after boot or after middlewared starts up for the first time.

Reply:
Yes but that would not happen here ;) This if would never be true twice